org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/services/jspServices.mtl

```diff
@@ -24,6 +24,11 @@
 [import org::eclipse::lyo::oslc4j::codegenerator::services::adaptorInterfaceServices/]
 [import org::eclipse::lyo::oslc4j::codegenerator::services::resourceServices/]
 [import org::eclipse::lyo::oslc4j::codegenerator::services::serviceProviderServices/]
+[import org::eclipse::lyo::oslc4j::codegenerator::services::fileServices/]
+
+[query public generateJspFile(anAdaptorInterface : AdaptorInterface, filePath : String) : Boolean = 
+    not (anAdaptorInterface.doNotRegenerateJspFiles()._and(fileExists(filePath)))
```

Comment:
**NB!** I think the logic proposed by @jposterloh was to refrain from generating code in `webapp-gen/a/b/c` if the developer already created `webapp/a/b/c`. So this requires a different target folder.